src/component.ts

```diff
@@ -23,13 +25,19 @@ export function componentFactory (
   Component: VueClass<Vue>,
   options: ComponentOptions<Vue> = {}
 ): VueClass<Vue> {
+  const reflectionMap: ReflectionMap = {
```

Comment:
Why do we need such temporary object and to process for it in different functions?  Why don't we just directly assign the original metadata to component properties?

Reply:
Do you mean ComponentOptions? If you mean that, it's a tricky way because i don't want to change options interface and expose internal work.

Reply:
No, I don't mean that. 
I mean why we need to store metadata in temporary object rather than just forward them directly on the place.
https://github.com/vuejs/vue-class-component/pull/227/files#diff-7db1d7c8dc88ce00d1aa236fa0c128ccR40
https://github.com/vuejs/vue-class-component/pull/227/files#diff-7db1d7c8dc88ce00d1aa236fa0c128ccR118

Reply:
Because forwarding code will be duplicated. If you okay with that, I'll do it in that way.

Reply:
Well, then we can extract the logic as a function. I believe we can remove `reflectionMap` without duplication.

something like:

```ts
function forwardMetadata(to: typeof Vue, from: typeof Vue, propertyKey: string): void {
  Reflect.getOwnMetadataKeys(from, propertyKey).forEach(metaKey => {
    const metadata = Reflect.getOwnMetadata(metaKey, from, propertyKey)
    Reflect.defineMetadata(metaKey, metadata, to, propertyKey)
  })
}
```

Then we just call it inline:

```ts
  Object.getOwnPropertyNames(Original).forEach(key => {
     // `prototype` should not be overwritten
     if (key === 'prototype') {
       return
     }
 	 
-    reflectionMap.static[key] = Reflect.getOwnMetadataKeys(Original, key);
+    forwardMetadata(Extended, Original, key)

     // Some browsers does not allow reconfigure built-in properties
     const extendedDescriptor = Object.getOwnPropertyDescriptor(Extended, key)
     if (extendedDescriptor && !extendedDescriptor.configurable) {
```

```ts
  forwardStaticMembersAndCollectReflection(Extended, Component, Super, reflectionMap)
-  copyReflectionMetadata(Component, Extended, reflectionMap)
+  Object.getOwnPropertyNames(proto).forEach(key => {
+    forwardMetadata(Extended.prototype, proto, key)
+  })
```